### PR TITLE
Add profiles for nodes and clusters

### PR DESCRIPTION
**Pull request by stgleb**
Extract common configurations to separate entities called profiles
that will enable to store them and provide to user e.g cpu intensive workload
or high memory capacity profile.

### CODE REVIEW

File: pkg/cloud_account/domain.go [hunk 1]
```diff
@@ -13,3 +15,12 @@ type CloudAccount struct {
 	Provider    provider.Name
 	Credentials Credentials
 }
+
+//Repository is used to abstract domain storage from implementation
+type Repository interface {
```
Comment (yegorch) [inline pkg/cloud_account/domain.go]: How come this interface is here? I've deleted it yesterday.

File: pkg/cloud_account/service.go [hunk 2]
```diff
@@ -3,73 +3,28 @@ package cloud_account
...
-
 	"github.com/pkg/errors"
-	"github.com/sirupsen/logrus"
-	"github.com/supergiant/supergiant/pkg/storage"
 )
 
-// Service holds all business logic related to cloud accounts
 type Service struct {
-	Repository storage.Interface
+	Repository Repository
```
Comment (yegorch) [inline pkg/cloud_account/service.go]: Should use storage.Interface

File: pkg/cloud_account/storage_integration_test.go [hunk 3]
```diff
@@ -0,0 +1,79 @@
+//+build integration
+
+package cloud_account
```
Comment (yegorch) [inline pkg/cloud_account/storage_integration_test.go]: Should be deleted already.

File: pkg/profile/kube_profile_service_test.go [hunk 4]
```diff
@@ -0,0 +1,147 @@
+package profile
+
+import (
+	"context"
+	"testing"
+
+	"github.com/pkg/errors"
+)
+
+type fakeStorage struct {
```
Comment (yegorch) [inline pkg/profile/kube_profile_service_test.go]: Why not using testify mocks?

File: pkg/profile/kube_profile.go [hunk 5]
```diff
@@ -0,0 +1,10 @@
+package profile
+
+type KubeProfile struct {
```
Comment (saheienko) [inline pkg/profile/kube_profile.go]: It is better to use a Kube, because exported names are qualified by their package names.

Comment (yegorch) [reply pkg/profile/kube_profile.go]: @saheienko Usually it is. But we already have Kube struct so it would be source of confusion

Comment (saheienko) [reply pkg/profile/kube_profile.go]: @yegorch Do you mean model.Kube?

File: pkg/profile/kube_profile_service_test.go (hunk 4 above)
Comment (stgleb) [reply pkg/profile/kube_profile_service_test.go]: Ok

File: pkg/profile/kube_profile.go (hunk 5 above)
Comment (yegorch) [reply pkg/profile/kube_profile.go]: @saheienko yes, model.Kube would be deprecated as everything in model package.

Comment (yegorch) [inline pkg/profile/kube_profile.go]: @stgleb add validations using govalidator

File: test/integration/profile/kube_profile_test.go [hunk 6]
```diff
@@ -0,0 +1,135 @@
...
+		t.Errorf("Unepexpected error while creating kube profile %v", err)
+	}
+
+	kube2, err := service.Get(ctx, kube.Id)
+
+	if err != nil {
+		t.Errorf("Unexpected error while getting kube profile %v", err)
+	}
+
+	if kube.Id != kube.Id || kube.KubernetesVersion != kube2.KubernetesVersion {
```
Comment (saheienko) [inline test/integration/profile/kube_profile_test.go]: kube.Id != kube.Id

File: test/integration/profile/node_profile_test.go [hunk 7]
```diff
@@ -0,0 +1,116 @@
...
+		t.Errorf("Unepexpected error while creating node profile %v", err)
+	}
+
+	node2, err := service.Get(ctx, node.Id)
+
+	if err != nil {
+		t.Errorf("Unexpected error while getting node profile %v", err)
+	}
+
+	if node.Id != node.Id || node.Image != node2.Image || node.Size != node2.Size {
```
Comment (saheienko) [inline test/integration/profile/node_profile_test.go]: node.Id != node.Id

**Comment by stgleb** [comment]
@gopherstein Will be done

**Comment by stgleb** [comment]
@TheKLARKEN I guess I've removed that in latest commits. If you found something else - please leave comments on the code.

**Pull request merged**